### ISSUE TITLE
embark--act args-out-of-range 0 0

### ISSUE THREAD

**Issue by aaronjensen** [oantolin/embark#149]
Hello again 😄 

Now I get this on some of my embark collects (but not all). Specifically when I do a:

```
  (defun aj/consult-ripgrep-in-directory ()
    "Find a directory and then search in that directory."
    (interactive)
    (consult-ripgrep t))
```

```
Debugger entered--Lisp error: (args-out-of-range 0 0)
  get-text-property(0 compilation-directory)
  #f(compiled-function (marker filename directory &rest formats) "Find a buffer for file FILENAME.\nIf FILENAME is not found at all, ask the user where to find it.\nPop up the buffer containing MARKER and scroll to MARKER if we ask\nthe user where to find the file.\nSearch the directories in `compilation-search-path'.\nA nil in `compilation-search-path' means to try the\n\"current\" directory, which is passed in DIRECTORY.\nIf DIRECTORY is relative, it is combined with `default-directory'.\nIf DIRECTORY is nil, that means use `default-directory'.\nFORMATS, if given, is a list of formats to reformat FILENAME when\nlooking for it: for each element FMT in FORMATS, this function\nattempts to find a file whose name is produced by (format FMT FILENAME)." #<bytecode 0x1f50b16ed17170a5>)(#<marker at 1 in  *temp*> "ws-internal_user_notification_component/package.sh" nil)
  apply(#f(compiled-function (marker filename directory &rest formats) "Find a buffer for file FILENAME.\nIf FILENAME is not found at all, ask the user where to find it.\nPop up the buffer containing MARKER and scroll to MARKER if we ask\nthe user where to find the file.\nSearch the directories in `compilation-search-path'.\nA nil in `compilation-search-path' means to try the\n\"current\" directory, which is passed in DIRECTORY.\nIf DIRECTORY is relative, it is combined with `default-directory'.\nIf DIRECTORY is nil, that means use `default-directory'.\nFORMATS, if given, is a list of formats to reformat FILENAME when\nlooking for it: for each element FMT in FORMATS, this function\nattempts to find a file whose name is produced by (format FMT FILENAME)." #<bytecode 0x1f50b16ed17170a5>) (#<marker at 1 in  *temp*> "ws-internal_user_notification_component/package.sh" nil))
  compilation-find-file-projectile-find-compilation-buffer(#f(compiled-function (marker filename directory &rest formats) "Find a buffer for file FILENAME.\nIf FILENAME is not found at all, ask the user where to find it.\nPop up the buffer containing MARKER and scroll to MARKER if we ask\nthe user where to find the file.\nSearch the directories in `compilation-search-path'.\nA nil in `compilation-search-path' means to try the\n\"current\" directory, which is passed in DIRECTORY.\nIf DIRECTORY is relative, it is combined with `default-directory'.\nIf DIRECTORY is nil, that means use `default-directory'.\nFORMATS, if given, is a list of formats to reformat FILENAME when\nlooking for it: for each element FMT in FORMATS, this function\nattempts to find a file whose name is produced by (format FMT FILENAME)." #<bytecode 0x1f50b16ed17170a5>) #<marker at 1 in  *temp*> "ws-internal_user_notification_component/package.sh" nil)
  apply(compilation-find-file-projectile-find-compilation-buffer #f(compiled-function (marker filename directory &rest formats) "Find a buffer for file FILENAME.\nIf FILENAME is not found at all, ask the user where to find it.\nPop up the buffer containing MARKER and scroll to MARKER if we ask\nthe user where to find the file.\nSearch the directories in `compilation-search-path'.\nA nil in `compilation-search-path' means to try the\n\"current\" directory, which is passed in DIRECTORY.\nIf DIRECTORY is relative, it is combined with `default-directory'.\nIf DIRECTORY is nil, that means use `default-directory'.\nFORMATS, if given, is a list of formats to reformat FILENAME when\nlooking for it: for each element FMT in FORMATS, this function\nattempts to find a file whose name is produced by (format FMT FILENAME)." #<bytecode 0x1f50b16ed17170a5>) (#<marker at 1 in  *temp*> "ws-internal_user_notification_component/package.sh" nil))
  #f(advice-wrapper :around #f(compiled-function (marker filename directory &rest formats) "Find a buffer for file FILENAME.\nIf FILENAME is not found at all, ask the user where to find it.\nPop up the buffer containing MARKER and scroll to MARKER if we ask\nthe user where to find the file.\nSearch the directories in `compilation-search-path'.\nA nil in `compilation-search-path' means to try the\n\"current\" directory, which is passed in DIRECTORY.\nIf DIRECTORY is relative, it is combined with `default-directory'.\nIf DIRECTORY is nil, that means use `default-directory'.\nFORMATS, if given, is a list of formats to reformat FILENAME when\nlooking for it: for each element FMT in FORMATS, this function\nattempts to find a file whose name is produced by (format FMT FILENAME)." #<bytecode 0x1f50b16ed17170a5>) compilation-find-file-projectile-find-compilation-buffer)(#<marker at 1 in  *temp*> "ws-internal_user_notification_component/package.sh" nil)
  apply(#f(advice-wrapper :around #f(compiled-function (marker filename directory &rest formats) "Find a buffer for file FILENAME.\nIf FILENAME is not found at all, ask the user where to find it.\nPop up the buffer containing MARKER and scroll to MARKER if we ask\nthe user where to find the file.\nSearch the directories in `compilation-search-path'.\nA nil in `compilation-search-path' means to try the\n\"current\" directory, which is passed in DIRECTORY.\nIf DIRECTORY is relative, it is combined with `default-directory'.\nIf DIRECTORY is nil, that means use `default-directory'.\nFORMATS, if given, is a list of formats to reformat FILENAME when\nlooking for it: for each element FMT in FORMATS, this function\nattempts to find a file whose name is produced by (format FMT FILENAME)." #<bytecode 0x1f50b16ed17170a5>) compilation-find-file-projectile-find-compilation-buffer) #<marker at 1 in  *temp*> "ws-internal_user_notification_component/package.sh" nil nil)
  ad-Advice-compilation-find-file(#f(advice-wrapper :around #f(compiled-function (marker filename directory &rest formats) "Find a buffer for file FILENAME.\nIf FILENAME is not found at all, ask the user where to find it.\nPop up the buffer containing MARKER and scroll to MARKER if we ask\nthe user where to find the file.\nSearch the directories in `compilation-search-path'.\nA nil in `compilation-search-path' means to try the\n\"current\" directory, which is passed in DIRECTORY.\nIf DIRECTORY is relative, it is combined with `default-directory'.\nIf DIRECTORY is nil, that means use `default-directory'.\nFORMATS, if given, is a list of formats to reformat FILENAME when\nlooking for it: for each element FMT in FORMATS, this function\nattempts to find a file whose name is produced by (format FMT FILENAME)." #<bytecode 0x1f50b16ed17170a5>) compilation-find-file-projectile-find-compilation-buffer) #<marker at 1 in  *temp*> "ws-internal_user_notification_component/package.sh" nil)
  apply(ad-Advice-compilation-find-file #f(advice-wrapper :around #f(compiled-function (marker filename directory &rest formats) "Find a buffer for file FILENAME.\nIf FILENAME is not found at all, ask the user where to find it.\nPop up the buffer containing MARKER and scroll to MARKER if we ask\nthe user where to find the file.\nSearch the directories in `compilation-search-path'.\nA nil in `compilation-search-path' means to try the\n\"current\" directory, which is passed in DIRECTORY.\nIf DIRECTORY is relative, it is combined with `default-directory'.\nIf DIRECTORY is nil, that means use `default-directory'.\nFORMATS, if given, is a list of formats to reformat FILENAME when\nlooking for it: for each element FMT in FORMATS, this function\nattempts to find a file whose name is produced by (format FMT FILENAME)." #<bytecode 0x1f50b16ed17170a5>) compilation-find-file-projectile-find-compilation-buffer) (#<marker at 1 in  *temp*> "ws-internal_user_notification_component/package.sh" nil))
  compilation-find-file(#<marker at 1 in  *temp*> "ws-internal_user_notification_component/package.sh" nil)
  apply(compilation-find-file #<marker at 1 in  *temp*> "ws-internal_user_notification_component/package.sh" nil nil)
  compilation-next-error-function(0 t)
  next-error(0 t)
  embark-goto-location("ws-internal_user_notification_component/package.sh...")
  funcall-interactively(embark-goto-location "ws-internal_user_notification_component/package.sh...")
  call-interactively(embark-goto-location nil nil)
  command-execute(embark-goto-location)
  #f(compiled-function () #<bytecode 0x3bf9ab57b05c7a8>)()
  embark--act(embark-goto-location "ws-internal_user_notification_component/package.sh...")
  embark-collect-choose(#<marker (moves after insertion) at 178 in *Embark Collect*<2>>)
  button-activate(#<marker (moves after insertion) at 178 in *Embark Collect*<2>> nil)
  push-button(178)
  funcall-interactively(push-button 178)
  call-interactively(push-button nil nil)
  command-execute(push-button)
```

**Comment by minad** [comment]
> next-error(0 t)

@oantolin Is this my fault and we should go back to the other function?

**Comment by oantolin** [comment]
I guess it is possible that this is whatever problem I ran into with `next-error` that I can't remember. @aaronjensen could you please test with this older definition:

```elisp
(defun embark-goto-location (location)
  "Go to LOCATION, which should be a string with a grep match."
  (interactive "sLocation: ")
  (with-temp-buffer
    (insert location "\n")
    (grep-mode)
    (goto-char (point-min))
    (compile-goto-error)))
```

Probably you'll still get the same error, but it'll be good to know for sure. I haven't been able to reproduce this error, but I did spot a problem: when I try the directory prompting call, `(consult-grep t)`, and pick a different directory, `embark-goto-location` does get confused about the path and asks me where the file is even though the collect buffer has the correct default directory. (This happens because actions always run in the the buffer that you happened to be in when you ran `consult-ripgrep`...)

**Comment by aaronjensen** [comment]
Same error for me with that version. I do have projectile in the mix, and it's in the trace. It's possibly erroring because it's trying to find a file outside of the current project.

**Comment by oantolin** [comment]
Well, let's try to determine if the paths are the problem. This might seem a little indirect, but try `M-: (setq embark--target-buffer nil embark--target-window nil)` in the collect buffer, and afterwards try pressing `RET` on some grep result.

**Comment by aaronjensen** [comment]
That works

**Comment by oantolin** [comment]
Great! So it seems like it is the default-directory. Let's try running embark-goto-location always in the window that you press `RET` in, instead of the target window. GitHub autoclosed this, because I said "attempt to fix #149", feel free to reopen if the attempt failed.

**Comment by aaronjensen** [comment]
That seemed to work for me, thank you again for the quick fix!
